l10n_eu_product_adr/models/adr_goods.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2021 Stefan Rijnhart <stefan@opener.amsterdam>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.osv.expression import AND
+
+from .common import TRANSPORT_CATEGORIES, TUNNEL_RESTRICTION_CODES
+
+
+class AdrGoods(models.Model):
+    _name = "adr.goods"
+    _description = "Dangerous Goods"
+
+    active = fields.Boolean(default=True)
+    un_number = fields.Char(
+        string="UN Number",
+        required=True,
+        help=(
+            "Contains the UN number: of the dangerous substance or article "
+            "if the substance or article has been assigned its own specific "
+            "UN number; or of the generic or n.o.s. entry to which the "
+            "dangerous substances or articles not mentioned by name shall "
+            "be assigned in accordance with the criteria ('decision trees') "
+            "of Part 2."
+        ),
+    )
+    name = fields.Char(
+        "Name and description",
+        required=True,
+        translate=True,
+        help=(
+            "Contains, in upper case characters, the name of the substance or "
+            "article, if the substance or article has been assigned its own "
+            "specific UN number, or of the generic or n.o.s. entry to which "
+            "it has been assigned in accordance with the criteria "
+            "('decision trees') of Part 2. This name shall be used as the "
+            "proper shipping name or, when applicable, as part of the proper "
+            "shipping name (see 3.1.2 for further details on the proper "
+            "shipping name)."
+        ),
+    )
+    class_id = fields.Many2one(comodel_name="adr.class", string="Class", required=True)
+    classification_code = fields.Char()
+    label_ids = fields.Many2many(
+        comodel_name="adr.label",
+        string="Labels",
+        help=(
+            "The labels/placards (see 5.2.2.2 and 5.3.1.7) that have to be affixed "
+            "to packages, containers, tank-containers, portable tanks, MEGCs and "
+            "vehicles.)"
+        ),
+    )
+    limited_quantity = fields.Char()
```

Comment:
This one should be different
```suggestion
    limited_quantity = fields.Float()
    limited_quantity_uom = fields.Many2one("uom.uom")
```

Reply:
OK. I will not commit your suggestion directly because of the implications, but I will have a look at this.

Reply:
See new commit "[IMP] l10n_eu_product_adr: distinct uom for limited quantity" in this PR for this.